test/pubsub.js

```diff
@@ -53,40 +47,35 @@ describe('pubsub', function () {
       let id1
       let id2
 
-      before('spawn nodes', function () {
+      before('spawn nodes', async function () {
         this.timeout(timeout)
 
-        return Promise.all(tests[name].map(fn => fn()))
-          .then(nodes => {
-            [daemon1, daemon2] = nodes
-          })
+        const nodes = await Promise.all(tests[name].map(fn => fn()))
```

Comment:
destruct?

Reply:
Somehow I'm getting a problem while destructuring in some cases. This is a good example:

It works ✅
```js
const nodes = await Promise.all(tests[name].map(fn => fn()))

console.log(nodes)
// Logs:
// [(...)]
```

It doesn't work ❌
```js
[daemon1, daemon2] = await Promise.all(tests[name].map(fn => fn()))

console.log(daemon1)
// Logs:
// undefined
console.log(daemon2)
// Logs:
// undefined
```

Am I doing something wrong?

Reply:
You need to add a semicolon between `this.timeout(timeout)` and the destructuring assignment.

E.g. this:

```javascript
this.timeout(timeout)

[daemon1, daemon2] = await Promise.all(tests[name].map(fn => fn()))
```

needs to become this:

```javascript
this.timeout(timeout);

[daemon1, daemon2] = await Promise.all(tests[name].map(fn => fn()))
```

otherwise the runtime thinks you're trying to do:

```javascript
this.timeout(timeout)[daemon1, daemon2] = await Promise.all(tests[name].map(fn => fn()))
```

and throws an error because `this.timeout()` doesn't return anything.

If it did it would try to assign an array with two members to the key `undefined` of the return value which is almost certainly not what you want:

```javascript

let val1, val2

async function main () {
  const val = {}

  function fn () {
    return val
  };

  fn()

  [val1, val2] = await Promise.all([1, 2])

  console.info(val)
  // prints "{ undefined: [ 1, 2 ] }"
}

main()
```

Anyway there's more info here: http://inimino.org/~inimino/blog/javascript_semicolons